jaeger/cmd/check.go

```diff
@@ -0,0 +1,226 @@
+			WithCheck(func(ctx context.Context) error {
+				// Check for Jaeger Service Account
+				_, err = kubeAPI.CoreV1().ConfigMaps(jaegerNamespace).Get(ctx, "collector-config", metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				return nil
+			}))
+
+	checkers = append(checkers,
+		*healthcheck.NewChecker("collector pod is running", "l5d-jaeger-collector-running", false, true, time.Time{}, false).
+			WithCheck(func(ctx context.Context) error {
+				// Check for Collector pod
+				podList, err := kubeAPI.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{LabelSelector: "component=collector"})
+				if err != nil {
+					return err
+				}
+				return checkPodsRunning(podList.Items)
+			}))
+
+	checkers = append(checkers,
+		*healthcheck.NewChecker("jaeger pod is running", "l5d-jaeger-jaeger-running", false, true, time.Time{}, false).
+			WithCheck(func(ctx context.Context) error {
+				// Check for Jaeger pod
+				podList, err := kubeAPI.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{LabelSelector: "component=jaeger"})
+				if err != nil {
+					return err
+				}
+				return checkPodsRunning(podList.Items)
+			}))
+
+	checkers = append(checkers,
+		*healthcheck.NewChecker("jaeger extension pods are injected", "l5d-jaeger-pods-injection", false, true, time.Time{}, false).
+			WithCheck(func(ctx context.Context) error {
+				// Check for Jaeger pod
+				pods, err := kubeAPI.GetPodsByNamespace(ctx, jaegerNamespace)
+				if err != nil {
+					return err
+				}
+				return checkIfDataPlanePodsExist(pods)
+			}))
+
+	return healthcheck.NewCategory(linkerdJaegerExtensionCheck, checkers, true), nil
+}
+
+func newCheckOptions() *checkOptions {
+	return &checkOptions{
+		wait:   300 * time.Second,
+		output: healthcheck.TableOutput,
+	}
+}
+
+func (options *checkOptions) validate() error {
+	if options.output != healthcheck.TableOutput && options.output != healthcheck.JSONOutput {
+		return fmt.Errorf("Invalid output type '%s'. Supported output types are: %s, %s", options.output, healthcheck.JSONOutput, healthcheck.TableOutput)
+	}
+	return nil
+}
+
+func newCmdCheck() *cobra.Command {
+	options := newCheckOptions()
+	cmd := &cobra.Command{
+		Use:   "check [flags]",
+		Args:  cobra.NoArgs,
+		Short: "Check the Jaeger extension for potential problems",
+		Long: `Check the jaeger extension for potential problems.
+
+The check command will perform a series of checks to validate that the Jaeger extension
+ is configured correctly. If the command encounters a
+failure it will print additional information about the failure and exit with a
+non-zero exit code.`,
+		Example: `  # Check that the Jaeger extension is up and running
+  linkerd jaeger check`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// Get jaeger Extension Namespace
+			ns, err := getNamespaceOfExtension(jaegerExtensionName)
+			if err != nil {
+				fmt.Fprint(os.Stderr, err.Error())
```

Comment:
I think this error string needs to have a newline at the end to render properly on the command line.

Reply:
Using `fmt.Fprintln` seems to help but this does not seem to be the case everywhere else.